### ISSUE TITLE
Multiselect FROM account(s) when sharing something through Mastalab from other apps such as newsreaders, browsers, etc.

### ISSUE THREAD

**Issue by LeeteqXV** [stom79/mastalab#24]
What I miss for real #multiaccount usage now is to get to select the FROM account when I choose to share something through Mastalab (as a new toot) from other apps such as newsreaders, browsers, etc.

btw, that selection of FROM account(S) should also reflect the #multiAccount feature; hence we should be able to select more than one account for each toot, just as we already do in #Tweetdeck .

IMO the interface could be the same for normal new toots as for new ones initiated from external apps, so that we can choose more than one account to toot  from/through in all/both scenarios. 

How about making the current active account avatar expandable as a "drop-down" menu just where it is now, which presents the other available/connected account avatars with checkboxes beside them?

**Comment by stom79** [comment]
OK. Second point will automatically solve the first one.
What I can do:
- Long click on the "Toot" button will suggest accounts (like it's done for boosts/favourites)
- I will add a checkbox at the left of each accounts (with only the authenticated account checked by default), thus several accounts can be selected for the action (toots/favourites/boosts).

Is it ok for you?

**Comment by LeeteqXV** [comment]
Yes, that sounds good. However, IMHO, the long-press function (which I personally like a lot) is not obvious or intuitive for new Mastalab users, so I think that the user avatar should have this option as well.

However, that newbie concern could possibly be dealt with through a "How to use Mastalab" multi-step introduction "slideshow" which can be autostarted when logging in to Mastalab for the first time (or for all new account logins?) (and also linked to from its About page/column to be able to find it back and repeat it later).

(Crossreferencing the original toot thread, in case more comments are coming in there: https://mastodon.host/@FuzboleroXV/98947756437145979 )

**Comment by LeeteqXV** [comment]
When coming from another app such as a news story in a newsreader or browser, and clicking on the standard Android "share" function, several apps including Mastalab appears in a list. When selecting Mastalab, I think that what should happen first is to be presented with a multi-select list of connected Mastodon accounts with optional checkboxes on one side, and the possibility for a single-tap directly on the account name when only wanting to select 1 account. If not tapping on the checkbox, but directly on a name, that should automatically take us inside Mastalab using that account.

Furthermore; IMO, AFTER such a single/multiaccount toot, there is one more concern related to efficiency:

- it would be very practical if Mastalab changed the "active" account to the account that was last used when coming in from an external app like that, BUT ONLY when using SINGLE-select. When using MultiSelect, I think that Mastalab should keep remembering the account that was previously used as the active one, not changing the active account to any of the multiselected ones.

This way it would be easy to get back into Mastalab with the desired account active when opening Mastalab manually later.

One more thing: I think that in the initial multi-select list, the ACTIVE account should be pre-selected for sub-sequent shares, again based on the last used single-select account. I am unsure as if it should remember and pre-select all lasts used multi-select accounts from last time or not. But I imagine that that would be a nice option to have in the settings.

**Comment by stom79** [comment]
"slideshow" is hard to maintain, especially with localization. That's why I started a [wiki](https://github.com/stom79/mastalab/wiki).
Thanks for other suggestions. I will first work on what I suggested and you validated. Later, I could add these other features.

**Comment by LeeteqXV** [comment]
I wrote "slideshow" with quotes, as I did not remember what it is called when showing "what's new" with prev/next" buttons on startup. Not necessarily a slideshow, just a multi-step information pane. (I think also Mastodon has done something like that with new releases, or is it just that it is so common elsewhere that my memory is mixed up?)

**Comment by iknownothingaboutcomputers** [comment]
@stom79 No need for checkboxes on one side. Use the same method which you use for fav/boosts and replies. Highlight the selected account and grey-out the ones not selected. Checkboxes will only clutter, imo.

**Comment by stom79** [comment]
I will see how to replace checkbox with highlighting. I might start with checkbox and then I will replace them with what you suggested.

**Comment by deutrino** [comment]
I think Android 7+ has a native feature that might be usable for this. My news reader app, inoreader, pops up a "share drawer" which I have seen in other new apps, and then after a slight delay, a second drawer pops further up with the destinations (mostly chat contacts) I share to most! Maybe there is a way to populate Mastodon accounts into this? @stom79 I will DM you a video, it has names of my contacts in it so I won't post it here. But maybe it will give you some ideas.
